contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -0,0 +1,365 @@
+}
+
+func (t *T) GetOfferId() string {
+	if t.Offer == nil {
+		return ""
+	}
+	return t.Offer.Details().Id.GetValue()
+}
+
+func generateTaskName(pod *api.Pod) string {
+	ns := pod.Namespace
+	if ns == "" {
+		ns = api.NamespaceDefault
+	}
+	return fmt.Sprintf("%s.%s.pods", pod.Name, ns)
+}
+
+func (t *T) BuildTaskInfo() *mesos.TaskInfo {
+	info := &mesos.TaskInfo{
+		Name:     proto.String(generateTaskName(&t.Pod)),
+		TaskId:   mutil.NewTaskID(t.ID),
+		SlaveId:  mutil.NewSlaveID(t.Spec.SlaveID),
+		Executor: t.executor,
+		Data:     t.Spec.Data,
+		Resources: []*mesos.Resource{
+			mutil.NewScalarResource("cpus", t.Spec.CPU),
+			mutil.NewScalarResource("mem", t.Spec.Memory),
+		},
+	}
+	if portsResource := rangeResource("ports", t.Spec.Ports); portsResource != nil {
+		info.Resources = append(info.Resources, portsResource)
+	}
+	return info
+}
+
+// Fill the Spec in the T, should be called during k8s scheduling,
+// before binding.
+func (t *T) FillFromDetails(details *mesos.Offer) error {
+	if details == nil {
+		//programming error
+		panic("offer details are nil")
+	}
+
+	log.V(3).Infof("Recording offer(s) %v against pod %v", details.Id, t.Pod.Name)
+
+	t.Spec = Spec{
+		SlaveID: details.GetSlaveId().GetValue(),
+		CPU:     containerCpus,
+		Memory:  containerMem,
+	}
+
+	if mapping, err := t.mapper.Generate(t, details); err != nil {
+		t.Reset()
+		return err
+	} else {
+		ports := []uint64{}
+		for _, entry := range mapping {
+			ports = append(ports, entry.OfferPort)
+		}
+		t.Spec.PortMap = mapping
+		t.Spec.Ports = ports
+	}
+
+	// hostname needs of the executor needs to match that of the offer, otherwise
+	// the kubelet node status checker/updater is very unhappy
+	const HOSTNAME_OVERRIDE_FLAG = "--hostname-override="
+	hostname := details.GetHostname() // required field, non-empty
+	hostnameOverride := HOSTNAME_OVERRIDE_FLAG + hostname
+
+	argv := t.executor.Command.Arguments
+	overwrite := false
+	for i, arg := range argv {
+		if strings.HasPrefix(arg, HOSTNAME_OVERRIDE_FLAG) {
+			overwrite = true
+			argv[i] = hostnameOverride
+			break
+		}
+	}
+	if !overwrite {
+		t.executor.Command.Arguments = append(argv, hostnameOverride)
+	}
+	return nil
+}
+
+// Clear offer-related details from the task, should be called if/when an offer
+// has already been assigned to a task but for some reason is no longer valid.
+func (t *T) Reset() {
+	log.V(3).Infof("Clearing offer(s) from pod %v", t.Pod.Name)
+	t.Offer = nil
+	t.Spec = Spec{}
+}
+
+func (t *T) AcceptOffer(offer *mesos.Offer) bool {
+	if offer == nil {
+		return false
+	}
+	var (
+		cpus float64 = 0
+		mem  float64 = 0
+	)
+	for _, resource := range offer.Resources {
+		if resource.GetName() == "cpus" {
+			cpus = *resource.GetScalar().Value
+		}
+
+		if resource.GetName() == "mem" {
+			mem = *resource.GetScalar().Value
+		}
+	}
+	if _, err := t.mapper.Generate(t, offer); err != nil {
+		log.V(3).Info(err)
+		return false
+	}
+	if (cpus < containerCpus) || (mem < containerMem) {
```

Comment:
Thanks for adding the TODO. In a separate PR can you please add something to 
https://github.com/mesosphere/kubernetes-mesos/blob/master/docs/issues.md
mentioning the two major differences between scheduling in k8sm and k8s that I'm aware of: resource-based scheduling and label selectors in pods (to constrain machine) don't work yet in k8sm. (Also add anything else you can think of).  Also, we really should upstream that whole documentation tree or at least provide a pointer from
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/mesos.md
to it.